lib/improver/standardise.py

```diff
@@ -0,0 +1,446 @@
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Plugin to regrid cube data and standardise metadata"""
+
+import warnings
+import numpy as np
+
+import iris
+from iris.analysis import Nearest, Linear
+from iris.exceptions import CoordinateNotFoundError
+from scipy.interpolate import griddata
+
+from improver import BasePlugin
+from improver.metadata.amend import amend_attributes
+from improver.metadata.check_datatypes import (
+    check_cube_not_float64, check_time_coordinate_metadata)
+from improver.metadata.constants.time_types import (
+    TIME_COORD_NAMES, TIME_REFERENCE_DTYPE, TIME_REFERENCE_UNIT,
+    TIME_DTYPE, TIME_UNIT)
+from improver.threshold import BasicThreshold
+from improver.utilities.cube_checker import spatial_coords_match
+from improver.utilities.spatial import OccurrenceWithinVicinity
+
+
+def grid_contains_cutout(grid, cutout):
+    """
+    Check that a spatial cutout is contained within a given grid
+
+    Args:
+        grid (iris.cube.Cube):
+            A cube defining a data grid
+        cutout (iris.cube.Cube):
+            The cutout to search for within the grid
+
+    Returns:
+        bool:
+            True if cutout is contained within grid, False otherwise
+    """
+    if spatial_coords_match(grid, cutout):
+        return True
+
+    # check whether "cutout" coordinate points match a subset of "grid"
+    # points on both axes
+    for axis in ['x', 'y']:
+        grid_coord = grid.coord(axis=axis)
+        cutout_coord = cutout.coord(axis=axis)
+        # check coordinate metadata
+        if (cutout_coord.name() != grid_coord.name() or
+                cutout_coord.units != grid_coord.units or
+                cutout_coord.coord_system != grid_coord.coord_system):
+            return False
+
+        # search for cutout coordinate points in larger grid
+        cutout_start = cutout_coord.points[0]
+        find_start = [np.isclose(cutout_start, grid_point)
+                      for grid_point in grid_coord.points]
+        if not np.any(find_start):
+            return False
+
+        start = find_start.index(True)
+        end = start + len(cutout_coord.points)
+        try:
+            if not np.allclose(cutout_coord.points,
+                               grid_coord.points[start:end]):
+                return False
+        except ValueError:
+            # raised by np.allclose if "end" index overshoots edge of grid
+            # domain - slicing does not raise IndexError
+            return False
+
+    return True
+
+
+class StandardiseGridAndMetadata(BasePlugin):
+
+    """Plugin to regrid cube data and standardise metadata"""
+
+    REGRID_REQUIRES_LANDMASK = {"bilinear": False,
+                                "nearest": False,
+                                "nearest-with-mask": True}
+
+    def __init__(self, regrid_mode='bilinear', extrapolation_mode='nanmask',
```

Comment:
I copied this over from the CLI, where `regrid_mode` is an optional argument for regridding.  I agree with you in principle, it might be better to change this so that the user has to specify it.  However doing that now would require quite a lot more changes to the CLI and bats test files, so I'd rather not do it as part of this (already very large) PR.

I will put something in the dostring though.